packages/front-end/components/DocLink.tsx

```diff
@@ -0,0 +1,100 @@
+import { ReactNode } from "react";
+
+const docLinks = {
```

Comment:
The naming doesn't feel very consistent or intuitive in this PR.  I'm proposing we use the following:

- **Link** represents the actual <a> tag
- **Section** represents a part of the documentation (what we're calling Key now)
- **Path** is the url path of the doc section (without the base url)
- **Url** is the full documentation url including the base

So, this variable would be named `docSections`, the type for the keys would be named `DocSection`.  Instead of `inferDocsLink` it would be `inferDocUrl`.  The DocLink component would be used like this:

```tsx
<DocLink section="mixpanel" fallback="/datasources">View Mixpanel Docs</DocLink>
```

Reply:
It also feels slightly weird for this mapping to mix URL paths (`/features`) and human-readable ids (`user_guide`).  Maybe the sections can all be human-readable ids and we can make a separate mapping for the inference part:

```tsx
const docSections = {
  features: "/app/features",
  experiments: "/app/experiments",
  metrics: "/app/metrics",
  ...
}

export type DocSection = keyof typeof docSections;

const urlMapping: Record<string, DocSection> = {
  "/features": "features",
  "/experiments": "experiments",
  "/experiment": "experiments",
  "/metrics": "metrics",
  "/metric": "metrics",
  ...
}
```